From 24ab23d15e8cb01b26f0c6205e7486617a56bb62 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Mon, 17 Feb 2020 14:20:15 -0700
Subject: [PATCH 28/28] bluetooth: Fix passkey uninitialized issue

This is the diff between v5 and v6 of the bluedump attack patchset:

https://lore.kernel.org/linux-bluetooth/20200214191609.Bluez.v5.1.Ia71869d2f3e19a76a6a352c61088a085a1d41ba6@changeid/
https://lore.kernel.org/linux-bluetooth/20200217120454.Bluez.v6.1.Ia71869d2f3e19a76a6a352c61088a085a1d41ba6@changeid/

Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 net/bluetooth/hci_event.c | 2 +-
 net/bluetooth/smp.c       | 9 ++++-----
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/net/bluetooth/hci_event.c b/net/bluetooth/hci_event.c
index 591e7477e925..e3942c96f54b 100644
--- a/net/bluetooth/hci_event.c
+++ b/net/bluetooth/hci_event.c
@@ -4562,7 +4562,7 @@ static void hci_user_confirm_request_evt(struct hci_dev *hdev,
 		 * legitimate or malicious.
 		 */
 		if (hci_find_link_key(hdev, &ev->bdaddr)) {
-			bt_dev_dbg(hdev, "Local host already has link key");
+			bt_dev_warn(hdev, "Local host already has link key");
 			confirm_hint = 1;
 			goto confirm;
 		}
diff --git a/net/bluetooth/smp.c b/net/bluetooth/smp.c
index 50e0ac692ec4..d0b13d65cfbc 100644
--- a/net/bluetooth/smp.c
+++ b/net/bluetooth/smp.c
@@ -2173,16 +2173,15 @@ static u8 smp_cmd_pairing_random(struct l2cap_conn *conn, struct sk_buff *skb)
 		if (smp->method != JUST_WORKS)
 			goto mackey_and_ltk;
 
-		/* If there already exists long term key in local host, leave
-		 * the decision to user space since the remote device could
-		 * be legitimate or malicious.
+		/* If there already exists link key in local host, leave the
+		 * decision to user space since the remote device could be
+		 * legitimate or malicious.
 		 */
 		if (hci_find_ltk(hcon->hdev, &hcon->dst, hcon->dst_type,
 				 hcon->role)) {
 			err = mgmt_user_confirm_request(hcon->hdev, &hcon->dst,
 							hcon->type,
-							hcon->dst_type,
-							passkey, 1);
+							hcon->dst_type, 0, 1);
 			if (err)
 				return SMP_UNSPECIFIED;
 			set_bit(SMP_FLAG_WAIT_USER, &smp->flags);
-- 
2.25.1

